### PR TITLE
Deprecate float values

### PR DESCRIPTION
**Pull request by w84thesun**
After [#1401](https://github.com/FerretDB/FerretDB/pull/1401)

Refs [#1151](https://github.com/FerretDB/FerretDB/issues/1151).

### CODE REVIEW

**Comment by chilagrow** [comment]
Tests are failing for FerretDB :thinking:

File: tests/diff/float_values_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		for name, tc := range map[string]struct {
+			doc      bson.D
+			expected mongo.CommandError
+		}{
+			"NaN": {
+				doc: bson.D{{"_id", "1"}, {"foo", math.NaN()}},
+				expected: mongo.CommandError{
+					Code:    2,
+					Name:    "BadValue",
+					Message: `wire.OpMsg.Document: validation failed for { _id: "1", foo: nan.0 } with: NaN is not supported`,
```
Comment (AlekSi) [inline tests/diff/float_values_test.go:43]: `nan.0`?

Comment (w84thesun) [reply tests/diff/float_values_test.go:43]: This output is from [FormatAnyValue](https://github.com/FerretDB/FerretDB/blob/d3cbadfa98be1c035dd318dafa9e5525f2aa9203/internal/types/path.go#L237). This is how MongoDB format it.

**Review by chilagrow — APPROVED** [review]

File: tests/diff/float_values_test.go (hunk 1 above)
Comment (rumyantseva) [reply tests/diff/float_values_test.go:43]: That's an interesting question: on the one hand, as we don't support `NaN`s anymore, we don't need to support that formatting, and we can offer something better than `nan.0` for `NaN`s. On the other hand, someone who has experience with MongoDB might expect to see `nan.0` in the response 🤔

File: tests/diff/float_values_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func TestFloatValues(t *testing.T) {
+	t.Parallel()
+
+	ctx, db := setup(t)
+
+	t.Run("Insert", func(t *testing.T) {
```
Comment (rumyantseva) [inline tests/diff/float_values_test.go:31]: This subtest is not necessary at the moment but might make sense in the future if decide to test not only inserts. So if you added in on purpose, let's keep it (and if you added it by mistake let's remove it).

**Review by rumyantseva — APPROVED** [review]
A couple of non-critical comments, and looks good to me otherwise.

**Review by noisersup — APPROVED** [review]
LGTM! Although we are not sure if it passes without merging FerretDB/FerretDB#1401

**Pull request merged**